src/api/app/models/bs_request.rb

```diff
@@ -150,11 +150,16 @@ def self.actions_summary(payload)
     ret.join(', ')
   end
 
-  def self.new_from_xml(xml)
+  def self.new_from_xml(xml, is_create: false)
     hashed = Xmlhash.parse(xml)
 
     raise SaveError, 'Failed parsing the request xml' unless hashed
 
+    if is_create
+      raise SaveError, 'Request ID attribute not allowed when creating a request' if hashed['id']
+      hashed.delete('state')
+    end
```

Comment:
Why does this happen in the model? There was good reason why Rails 4 moved mass assignment protection from the model to the controller.

> The whole point of the controller is to control the flow between user and application, including authentication, authorization, and as part of that access control.

So this should be the responsibility of the controller IMHO!

Reply:
@ChrisBr:

> Why does this happen in the model? There was good reason why Rails 4 moved mass assignment protection from the model to the controller.
> 
> > The whole point of the controller is to control the flow between user and application, including authentication, authorization, and as part of that access control.
> 
> So this should be the responsibility of the controller IMHO!

I will move the code to the controller.

Reply:
Just brain storming regarding your questions this morning:
1) You could make this function to accept both, string and xml hash (although not super nice)
2) It seems like xml gets always wrapped into an xml hash and stored in the params hash, so no need to parse it again (https://github.com/openSUSE/open-build-service/blob/master/src/api/config/initializers/wrap_parameters.rb#L61)

Reply:
@ChrisBr, code moved. Please have a look again.

> It seems like xml gets always wrapped into an xml hash and stored in the params hash, so no need to parse it again (https://github.com/openSUSE/open-build-service/blob/master/src/api/config/initializers/wrap_parameters.rb#L61)

Ops. To use the `xmlhash` parameter seems the right approach. I wil try it.